### PR TITLE
[docs-only] add release roadmap

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
Add add doc sections about our Release Roadmap. This was considered as needed because the confusion about our 1.0.0 Tech Preview release. It tries to clarify what to expect from major version 1 and why we do things like we do.

## Motivation and Context
Make clear what we do and why we do it like this.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: docs/ocis/release_roadmap.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+It isn't our intention to scare you with our addendum "Tech Preview". We want you to have a clear picture of what you can expect from oCIS. You could take it as a disclaimer or even compare it like running an Linux kernel in alpha stage. It can be very pleasing to be on the latest codebase but you could also find yourself with a lot of problems arising because of that.
+
+You clearly can expect a totally new experience of file-sync and share with oCIS and want you to use it now - but with understanding and caution.
```
Comment (phil-davis) [inline docs/ocis/release_roadmap.md:26]:
```suggestion
You clearly can expect a totally new experience of file-sync and share with oCIS and we want you to use it now - but with understanding and caution.
```

Comment (phil-davis) [reply docs/ocis/release_roadmap.md:26]: Or maybe:
```suggestion
You clearly can expect a totally new experience of file-sync and share with oCIS and we want you to try it now - but with understanding and caution.
```
What level between "try" and "use" to we want to suggest is "reasonable"?

File: docs/ocis/release_roadmap.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## ownCloud Infinite Scale 2.x general availability releases
+
+Starting with oCIS 2.0.0 release we will strictly stick to SemVer, just as we do right now for ownCloud Server. The general availability release also means that we can recommend oCIS warmly to everyone. Use it to store your precious family pictures or you confidential company data!
```
Comment (phil-davis) [inline docs/ocis/release_roadmap.md:30]:
```suggestion
Starting with oCIS 2.0.0 release we will strictly stick to SemVer, just as we do right now for ownCloud Server. The general availability release will also mean that we can recommend oCIS warmly to everyone. Use it to store your precious family pictures or you confidential company data!
```

File: docs/ocis/release_roadmap.md (hunk 1 above)
Comment (wkloucek) [reply docs/ocis/release_roadmap.md]: I think at some point we want people more than trying it out once. We want them to use it. But like in my example with the alpha stage kernel they should feel confident about what they are doing and understanding the involved risks.

**Comment by wkloucek** [comment]
@dragotin have a look ;-)

**Comment by wkloucek** [comment]
@dragotin please have a look

**Review by dragotin — APPROVED** [review]
Very good.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**